catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -0,0 +1,303 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source.solr;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.solr.client.solrj.SolrClient;
+import org.codice.ddf.configuration.PropertyResolver;
+import org.codice.solr.factory.impl.ConfigurationStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.ContentType;
+import ddf.catalog.filter.FilterAdapter;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceMonitor;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.util.impl.MaskableImpl;
+
+public abstract class RemoteSolrCatalogProvider extends MaskableImpl implements CatalogProvider {
+
+    private static final String PING_ERROR_MESSAGE = "Solr ping failed.";
+
+    private static final String OK_STATUS = "OK";
+
+    private static final String DESCRIBABLE_PROPERTIES_FILE = "/describable.properties";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoteSolrCatalogProvider.class);
+
+    private static Properties describableProperties = new Properties();
+
+    static {
+        try (InputStream inputStream = RemoteSolrCatalogProvider.class.getResourceAsStream(
+                DESCRIBABLE_PROPERTIES_FILE)) {
+            describableProperties.load(inputStream);
+        } catch (IOException e) {
+            LOGGER.info("Did not load properties properly.", e);
+        }
+    }
+
+    protected static final String SOLR_CATALOG_CORE_NAME = "catalog";
+
+    private String url;
+
+    private CatalogProvider provider = new UnavailableSolrCatalogProvider();
+
+    private SolrClient client;
+
+    private FilterAdapter filterAdapter;
+
+    private SolrFilterDelegateFactory solrFilterDelegateFactory;
+
+    private DynamicSchemaResolver resolver;
+
+    private Future<SolrClient> clientFuture;
+
+    /**
```

Comment:
For the sake of being consistent, should we just remove this JavaDoc comment, or improve it and add comments on the other constructors?